### DISCUSSION TITLE
How to submit to App Store Connect

### DISCUSSION THREAD

**Discussion by anthonydito** [bazelbuild/rules_apple#1744] [Q&A] [answered]
Hi all, I have got to the phase of my project where I would like to submit my app to App Store Connect for TestFlight.

However, I have been struggling to get an IPA signed correctly.

The current error that I am running into is:

```
Missing Provisioning Profile - Apps must contain a provisioning profile in a file named embedded.mobileprovision.
```

I am wondering what process others here use to turn your ios_application into something that you can submit to Apple.

**Comment by keith** [comment]
Have you set the `provisioning_profile` attribute on your `ios_application` rule?

**Reply by anthonydito** [reply]
I have. Here is my ios_application for reference

```
load("@build_bazel_rules_apple//apple:ios.bzl", "ios_application")

ios_application(
    name="ios",
    bundle_id="com.ditotechnologies.fourier",
    bundle_name="fourier",
    families=[
        "iphone",
        "ipad"
    ],
    minimum_os_version = "15.0",
    launch_storyboard="Launch.storyboard",
    infoplists = ["Info.plist"],
    app_icons = glob(["Icons.xcassets/**"], exclude=["**/.DS_Store"]),
    strings=["Localizable.strings"],
    provisioning_profile="fourier_prod.mobileprovision",
    resources=[
        "//ios/FourierMetalRendering:FourierMetalRenderingShaders"
    ],
    deps=[
        "//ios/Entrypoint:Entrypoint",
    ]
)
```

**Reply by keith** [reply]
What's the specific command you're using to build? Does the produced ipa have the file from the error? What's the output of `codesign -dvvv path/to/app`?

**Reply by anthonydito** [reply]
> What's the specific command you're using to build? 

```
bazel build ios --ios_signing_cert_name "Apple Development: xxx" --apple_generate_dsym
```

> Does the produced ipa have the file from the error?

No, the `embedded.mobileprovision` does not exist in the ipa

> What's the output of codesign -dvvv path/to/app?

Get the result of "code object is not signed at all"

Assuming, that the codesign is the issue here?

**Reply by keith** [reply]
it sounds like you're not building for device? are you passing `--ios_multi_cpus=arm64` anywhere to specify that?

**Reply by anthonydito** [reply]
Yep, was not built for device. Now, I get new errors that I think I will be able to resolve with some trial and error, thanks for the help!

Was using Tulsi for my day-to-day work so didn't know that I needed this flag to build for submission to app store connect.

**Comment by anthonydito** [comment, marked as answer]
If anybody finds this useful, here are the things that I needed to do to get a project that I had developed using Tulsi into app store connect.

1) Added these keys into my Info.plist
```
	<key>CFBundlePackageType</key>
	<string>APPL</string>
	<key>UIRequiredDeviceCapabilities</key>
	<array>
		<string>arm64</string>
	</array>
```

2. Build the app with Bazel 

`bazel build //ios:ios_prod --apple_generate_dsym --ios_multi_cpus=arm64 --define=apple.add_debugger_entitlement=no`

> Note, you need to use a production provisioning profile. I ended up creating a whole new `ios_application` that specifies this provisioning profile instead of my development.

3. Deploy to App Store Connect

`xcrun altool --upload-app --file your_path_to_ipa  -u your_username -p your_password --type ios`

> Note, for the password, you need to get a application specific password on your apple account.
